packages/next/server/config.ts

```diff
@@ -45,9 +46,8 @@ const experimentalWarning = execOnce(
   }
 )
 
-export function setHttpAgentOptions(
-  options: NextConfigComplete['httpAgentOptions']
-) {
+export function setHttpClientAndAgentOptions(options: NextConfig) {
+  ;(global as any).__NEXT_USE_UNDICI = options.experimental?.useUndici
```

Comment:
As discussed @Ethan-Arrowood let's wrap this and check if `process.version` is at least `v16.8.0` as pointed out in https://github.com/vercel/next.js/pull/40318#pullrequestreview-1099498389

We should also warn if the developer is setting this flag with a lower Node.js version

Reply:
Yes, it is not ideal, but fetch will only work in Node 16.8+ because of the APIs it depends on from Node.js itself. See this PR https://github.com/nodejs/undici/pull/1534 that expands on this